lib/pmdtester/builders/summary_report_builder.rb

```diff
@@ -0,0 +1,146 @@
+require_relative './html_report_builder'
+require_relative '../pmd_branch_detail'
+
+module PmdTester
+  # Building summary report to show the details about projects and pmd branchs
+  class SummaryReportBuilder < HtmlReportBuilder
+    REPORT_DIR = 'target/reports/diff'.freeze
+    BASE_CONFIG_PATH = 'target/reports/diff/base_config.xml'.freeze
+    PATCH_CONFIG_PATH = 'target/reports/diff/patch_config.xml'.freeze
+    INDEX_PATH = 'target/reports/diff/index.html'.freeze
+
+    def build(projects, base_name, patch_name)
+      @projects = projects
+      @base_details = get_branch_details(base_name)
+      @patch_details = get_branch_details(patch_name)
+
+      FileUtils.mkdir_p(REPORT_DIR) unless File.directory?(REPORT_DIR)
+      index = File.new(INDEX_PATH, 'w')
+
+      html_report = build_html_report('Summary report')
+      copy_css(REPORT_DIR)
+
+      index.puts html_report
+      index.close
+
+      puts 'Built summary report successfully!'
+    end
+
+    def get_branch_details(branch_name)
+      details = PmdBranchDetail.new(branch_name)
+      details.load
+      details
+    end
+
+    def build_body(doc)
+      build_branch_details_section(doc)
+      build_projects_section(doc)
+    end
+
+    def build_branch_details_section(doc)
+      doc.div(class: 'section', name: 'Branch details') do
```

Comment:
The name attribute should be the id. And just a personal preference of me: I tend to use only lower-case letters for the ids and avoid using any special characters and spaces.

Reply:
Done.